osh/worker/csmock_runner.py

```diff
@@ -100,19 +100,18 @@ def do(self, command, output_path=None, su_user=None, **kwargs):
         # ending -results.tar.xz, which appears second in the glob results
         return glob_results[-1], retcode
 
-    def analyze(self, analyzers, srpm_path, profile=None, su_user=None, additional_arguments=None,
-                result_filename=None, profile_url=None, **kwargs):
+    def determine_output_path(self, srpm_path, result_filename):
         if result_filename is None:
             result_filename = os.path.basename(srpm_path)[:-8]
-        if self.tmpdir:
-            output_path = os.path.join(self.tmpdir, result_filename + '.tar.xz')
-        else:
-            output_path = os.path.join(os.getcwd(), result_filename + '.tar.xz')
+        output_path = os.path.join(self.tmpdir if self.tmpdir else os.getcwd(),
+                                   result_filename + '.tar.xz')
```

Comment:
The `result_filename` maybe `None`, so we cannot use `None` + '.tar.xz'

Reply:
it won't because L105 makes sure it's not. But this indeed remind me of an issue, i.e., 
```
    def determine_output_path(self, srpm_path, result_filename):
        if result_filename is None:
            result_filename = os.path.basename(srpm_path)[:-8]
        output_path = os.path.join(self.tmpdir if self.tmpdir else os.getcwd(),
                                   result_filename + '.tar.xz')
```
I guess the slicing(`[:8]`) is to remove the trailing `.src.rpm` in the filename. In case of dist-git url, The commit hash would be truncated. I'll improve this.

Reply:
I had a second check, it won't be truncated since `result_filename` is processed elsewhere:
https://github.com/openscanhub/openscanhub/pull/278/files#diff-2ae41bcd4e6cf6150fb0c793f090773acb92488f9922cfae2529518523ab43e0R432-R434